freqtrade/optimize/hyperopt.py

```diff
@@ -237,27 +237,63 @@ def init_spaces(self):
             logger.debug("Hyperopt has 'protection' space")
             # Enable Protections if protection space is selected.
             self.config['enable_protections'] = True
-            self.protection_space = self.custom_hyperopt.protection_space()
+            try:
+                self.protection_space = self.custom_hyperopt.protection_space()
+            except OperationalException as e:
+                if self.config["hyperopt_ignore_unparam_space"]:
+                    logger.warning(e)
+                else:
+                    raise
 
         if HyperoptTools.has_space(self.config, 'buy'):
             logger.debug("Hyperopt has 'buy' space")
-            self.buy_space = self.custom_hyperopt.buy_indicator_space()
+            try:
+                self.buy_space = self.custom_hyperopt.buy_indicator_space()
+            except OperationalException as e:
+                if self.config["hyperopt_ignore_unparam_space"]:
```

Comment:
you MUST use `self.config.get("hyperopt_ignore_unparam_space", False):` here (obviously also below) - otherwise this will fail when the option is not given (this is what's causing tests to fail)